utils/test/test_titlecase_titles.exs

```diff
@@ -0,0 +1,72 @@
+defmodule MyTitleCaseTest do
```

Comment:
I attempted this to see if I could simplify it. What do you think?

```elixir
     exercises = fetch_livebooks("../exercises/")
    reading = fetch_livebooks("../reading/")

    Enum.map(exercises ++ reading ++ ["../start.livemd"], fn file_name ->
      File.stream!(file_name)
      |> Stream.with_index()
      |> Stream.map(fn
        {line, 0} ->
          refute Regex.match?(~r/\#\s([a-z]\w+)+/, line),
                 "#{file_name}:1 Heading should be capitalized"

        {line, index} ->
          refute Regex.match?(~r/\#\#\s([a-z]\w+)+/, line),
                 "#{file_name}:#{index + 1} Heading should be capitalized"
      end)
      |> Stream.run()
    end)
```

We could also make a script to automatically fix these issues:
I didn't add automatically fixing heading 1s because that doesn't seem as tedious to solve manually IMO.

```elixir
    exercises = fetch_livebooks("../exercises/")
    reading = fetch_livebooks("../reading/")

    Enum.map(exercises ++ reading ++ ["../start.livemd"], fn file_name ->
      file = File.read!(file_name)

      updated_file =
        Regex.replace(~r/(\#\#\s)(.+)/, file, fn _, heading_tag, heading ->
          capitalized_heading =
            String.split(heading)
            |> Enum.map(&String.capitalize/1)
            |> Enum.join(" ")

          heading_tag <> capitalized_heading
        end)

      File.write!(file_name, updated_file)
    end)
```

Reply:
Also, I think this test belongs in `utils_test.exs`

Reply:
Totally agree on simplifying the nested ifs. I can work on this.

On your first example (the non-script one), I tried that regex and the test fails where the headline is method name like `### increased_by/1` in [`measurements.livemd`](https://github.com/DockYard-Academy/beta_curriculum/blob/c10146906fec0c37430c64cfca8d8bf6090786ed/exercises/measurements.livemd?plain=1#L47) or in `## BookController.edit/2` in [`book_search_books.livemd`](https://github.com/DockYard-Academy/beta_curriculum/blob/c10146906fec0c37430c64cfca8d8bf6090786ed/reading/book_search_books.livemd?plain=1#L1121). My solution to this was just filter those out with 
```
~r/.*\/[0-9]$|[!"#$%&'()*+,-.:;<=>?@[\]^_`{|}~]|#+|\b[0-9]+\b/
```
which takes care of method names, words with punctuation in them and words with numbers.

Also there are 500+ `h3`s and 4 `h4`s in the codebase, do we want to exclude those since the regex targes only h1s and h2s?

On the use `Stream`, I'm still new to Elixir, is that an optimization so we don't have to load the whole file into memory? 

Thanks for looking into this!

Reply:
I think I like the AST approach.  Nevertheless, let's go ahead and merge this in if you're ok with it.